### PR TITLE
Cisco Umbrella Reporting - add new plugin

### PR DESCRIPTION
**Pull request by r7-mgorny**
## Proposed Changes
Cisco Umbrella Reporting - add new plugin

### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Umbrella Reporting:1.0.0. Step name: get_domain_visits\n",
    "meta": {},
    "output": {
      "domain_visits": [
        {
          "actionTaken": "BLOCKED",
          "categories": [
            "Dynamic DNS",
            "Web Hosting"
          ],
          "datetime": "2020-11-06T00:04:30.328Z",
          "destination": "rapid7.com",
          "externalIp": "198.51.100.100",
          "internalIp": "198.51.100.100",
          "originId": 121212121,
          "originLabel": "User (user@example.com)",
          "originType": "Anyconnect Roaming Client",
          "tags": []
        },
        {
          "actionTaken": "BLOCKED",
          "categories": [
            "Dynamic DNS",
            "Web Hosting"
          ],
          "datetime": "2020-11-06T00:04:30.328Z",
          "destination": "rapid7.com",
          "externalIp": "198.51.100.100",
          "internalIp": "198.51.100.100",
          "originId": 121212122,
          "originLabel": "User (user@example.com)",
          "originType": "Anyconnect Roaming Client",
          "tags": []
        }
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_umbrella_reporting:1.0.0 --debug run < tests/get_domain_visits.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Umbrella Reporting:1.0.0. Step name: get_domain_visits\n",
    "meta": {},
    "output": {
      "domain_visits": [
        {
          "actionTaken": "BLOCKED",
          "categories": [
            "Software/Technology",
            "Malware"
          ],
          "datetime": "2020-11-06T00:04:30.328Z",
          "destination": "rapid7.com",
          "externalIp": "198.51.100.100",
          "internalIp": "198.51.100.100",
          "originId": 121212123,
          "originLabel": "User (user@example.com)",
          "originType": "Anyconnect Roaming Client",
          "tags": []
        },
        {
          "actionTaken": "BLOCKED",
          "categories": [
            "Software/Technology",
            "Malware"
          ],
          "datetime": "2020-11-06T00:04:30.328Z",
          "destination": "rapid7.com",
          "externalIp": "198.51.100.100",
          "internalIp": "198.51.100.100",
          "originId": 121212124,
          "originLabel": "User (user@example.com)",
          "originType": "Anyconnect Roaming Client",
          "tags": []
        }
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_umbrella_reporting:1.0.0 --debug run < tests/get_security_activity.json
</summary>
</details>

### Test

Autogenerate with:
<details>

```
{
  "body": {
    "error": "Connection test failed!\n\nInvalid username or password provided. Verify your username and password are correct.",
    "log": "Connect: Connecting...\nrapid7/Cisco Umbrella Reporting:1.0.0. Step name: get_domain_visits\nConnection test failed!\n\nInvalid username or password provided. Verify your username and password are correct.\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/site-packages/cisco_umbrella_reporting_rapid7_plugin-1.0.0-py3.8.egg/icon_cisco_umbrella_reporting/connection/connection.py\", line 24, in test\n    self.client.security_activity_report()\n  File \"/usr/local/lib/python3.8/site-packages/cisco_umbrella_reporting_rapid7_plugin-1.0.0-py3.8.egg/icon_cisco_umbrella_reporting/util/api.py\", line 25, in security_activity_report\n    return self._call_api(\n  File \"/usr/local/lib/python3.8/site-packages/cisco_umbrella_reporting_rapid7_plugin-1.0.0-py3.8.egg/icon_cisco_umbrella_reporting/util/api.py\", line 45, in _call_api\n    raise PluginException(preset=PluginException.Preset.USERNAME_PASSWORD)\ninsightconnect_plugin_runtime.exceptions.PluginException: An error occurred during plugin execution!\n\nInvalid username or password provided. Verify your username and password are correct.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.2.0-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 326, in handle_step\n    output = self.start_step(\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.2.0-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 481, in start_step\n    output = func()\n  File \"/usr/local/lib/python3.8/site-packages/cisco_umbrella_reporting_rapid7_plugin-1.0.0-py3.8.egg/icon_cisco_umbrella_reporting/connection/connection.py\", line 29, in test\n    raise ConnectionTestException(\ninsightconnect_plugin_runtime.exceptions.ConnectionTestException: Connection test failed!\n\nInvalid username or password provided. Verify your username and password are correct.\n",
    "meta": {},
    "status": "error"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_umbrella_reporting:1.0.0 --debug test < tests/test_connection_bad.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Umbrella Reporting:1.0.0. Step name: get_domain_visits\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_umbrella_reporting:1.0.0 --debug test < tests/get_domain_visits.json
</summary>
</details>

### Validate

Autogenerate with:
<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Plugin failed validation! The following validation errors occurred:

Validator "IconValidator" failed!
        Cause: icon.png file not included in plugin.


----
[*] Total time elapsed: 927.295ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
[SUCCESS] Passes flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.3
Run started:2020-11-06 01:09:07.106240

Test results:
>> Issue: [B105:hardcoded_password_string] Possible hardcoded password: 'api_secret'
   Severity: Low   Confidence: Medium
   Location: ./icon_cisco_umbrella_reporting/connection/schema.py:8
   More Info: https://bandit.readthedocs.io/en/latest/plugins/b105_hardcoded_password_string.html
7           API_KEY = "api_key"
8           API_SECRET = "api_secret"
9           ORGANIZATION_ID = "organization_id"

--------------------------------------------------

Code scanned:
        Total lines of code: 313
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 1.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 1.0
                High: 0.0
Files skipped (0):
[FAIL] Fails bandit security checks

[*] Checking for typos with Misspell...
[SUCCESS] Passes Misspell check

```

<summary>
make validate
</summary>
</details>

Autogenerate with:
<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
[*] Plugin failed validation! The following validation errors occurred:

Validator "IconValidator" failed!
        Cause: icon.png file not included in plugin.


----
[*] Total time elapsed: 14750.287ms

```

<summary>
icon-validate --all .
</summary>
</details>

### CODE REVIEW

File: cisco_umbrella_reporting/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+FROM rapid7/insightconnect-python-3-38-plugin:4
```
Comment (mrinehart-r7) [inline cisco_umbrella_reporting/Dockerfile:1]:
```suggestion
FROM rapid7/insightconnect-python-3-38-slim-plugin:4
```

File: cisco_umbrella_reporting/help.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
+# Description
+
+This plugin utilizes Cisco Umbrella to get the most complete view of the relationships and evolution of Internet domains, IP addresses, and autonomous systems to pinpoint attackers infrastructures and predict future threats
```
Comment (mrinehart-r7) [inline cisco_umbrella_reporting/help.md:3]:
```suggestion
This plugin utilizes Cisco Umbrella to get the most complete view of the relationships and evolution of internet domains, IP addresses, and autonomous systems to pinpoint attackers infrastructures and predict future threats
```

File: cisco_umbrella_reporting/help.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+# Key Features
+
+* Most Recent Requests
+* Security Activity Report
+
+# Requirements
+
+* Requires an API Key and Secret from the Cisco Umbrella
```
Comment (mrinehart-r7) [inline cisco_umbrella_reporting/help.md:12]:
```suggestion
* Requires an API Key and Secret from Cisco Umbrella
```

File: cisco_umbrella_reporting/help.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+* Requires organization ID
```
Comment (mrinehart-r7) [inline cisco_umbrella_reporting/help.md:13]:
```suggestion
* Requires organization ID
```
Which organization ID is this referring to? Cisco Umbrella? This should be specified, as a user may confuse it with a Rapid7 organization ID.

File: cisco_umbrella_reporting/icon_cisco_umbrella_reporting/actions/get_domain_visits/action.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+            description=Component.DESCRIPTION,
+            input=GetDomainVisitsInput(),
+            output=GetDomainVisitsOutput())
+
+    def run(self, params={}):
+        if params.get(Input.MOST_RECENT, True):
+            if not params.get(Input.DOMAIN):
+                raise PluginException(
+                    cause="Input error.",
+                    assistance="When input 'Most Recent' is true domain input is required."
```
Comment (mrinehart-r7) [inline cisco_umbrella_reporting/icon_cisco_umbrella_reporting/actions/get_domain_visits/action.py:21]:
```suggestion
                    assistance="The 'domain' input is required when 'Most Recent' input is true."
```

File: cisco_umbrella_reporting/icon_cisco_umbrella_reporting/util/api.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import json
+import requests
+from requests.auth import HTTPBasicAuth
+from insightconnect_plugin_runtime.exceptions import PluginException
+
```
Comment (mrinehart-r7) [inline cisco_umbrella_reporting/icon_cisco_umbrella_reporting/util/api.py:5]:
```suggestion
from logging import Logger
from typing import Optional
```

File: cisco_umbrella_reporting/icon_cisco_umbrella_reporting/util/api.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+class CiscoUmbrellaReportingAPI:
+    def __init__(self, api_key: str, api_secret: str, organization_id: str, logger: object):
```
Comment (mrinehart-r7) [inline cisco_umbrella_reporting/icon_cisco_umbrella_reporting/util/api.py:8]:
```suggestion
    def __init__(self, api_key: str, api_secret: str, organization_id: str, logger: Logger):
```

File: cisco_umbrella_reporting/icon_cisco_umbrella_reporting/util/api.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    def security_activity_report(self) -> dict:
+        return self._call_api(
+            "GET",
+            f"organizations/{self.organization_id}/security-activity",
+            {
+                "limit": 500
+            }
+        )
+
+    def _call_api(self, method: str, path: str, json_data: dict = None, params: dict = None) -> dict:
```
Comment (mrinehart-r7) [inline cisco_umbrella_reporting/icon_cisco_umbrella_reporting/util/api.py:33]:
```suggestion
    def _call_api(self, method: str, path: str, json_data: Optional[dict] = None, params: Optional[dict] = None) -> dict:
```

File: cisco_umbrella_reporting/plugin.spec.yaml [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+- umbrella
+- reporting
+hub_tags:
+  use_cases: [threat_detection_and_response]
+  keywords: [cisco, umbrella, reporting]
+  features: []
+connection:
+  api_key:
+    title: API Key
+    description: Enter API key
```
Comment (mrinehart-r7) [inline cisco_umbrella_reporting/plugin.spec.yaml:28]:
```suggestion
    description: Cisco Umbrella API key
```

File: cisco_umbrella_reporting/plugin.spec.yaml [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+    type: credential_secret_key
+    required: true
+    example: 9de5069c5afe602b2ea0a04b66beb2c0
+  api_secret:
+    title: API Secret Key
+    description: Enter secret key
```
Comment (mrinehart-r7) [inline cisco_umbrella_reporting/plugin.spec.yaml:34]:
```suggestion
    description: Cisco Umbrella API secret key
```

File: cisco_umbrella_reporting/plugin.spec.yaml [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+    type: credential_secret_key
+    required: true
+    example: 9de5069c5afe602b2ea0a04b66beb2c0
+  organization_id:
+    title: Organization ID
+    description: ID of your organization
```
Comment (mrinehart-r7) [inline cisco_umbrella_reporting/plugin.spec.yaml:40]:
```suggestion
    description: ID of your Cisco Umbrella organization
```
Specify Cisco Umbrella or else it may cause confusion to users :)

File: cisco_umbrella_reporting/plugin.spec.yaml [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+      type: string
+      required: false
+    externalIp:
+      title: External IP
+      description: External IP address of the identity making the request
+      type: string
+      required: false
+    originLabel:
+      title: Origin Label
+      description: Human readable name for the identity, matches the one seen in the dashboard
```
Comment (mrinehart-r7) [inline cisco_umbrella_reporting/plugin.spec.yaml:63]:
```suggestion
      description: Human-readable name for the identity, matching the one seen in the dashboard
```

File: cisco_umbrella_reporting/plugin.spec.yaml [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+      required: false
+    tags:
+      title: Tags
+      description: Which tags, if any, the destination for which this request was made falls into
+      type: "[]string"
+      required: false
+actions:
+  get_domain_visits:
+    title: Get Domain Visits
+    description: Get a list of computers that visited a domain for the last 24 hours
```
Comment (mrinehart-r7) [inline cisco_umbrella_reporting/plugin.spec.yaml:99]:
```suggestion
    description: Get a list of computers that visited a domain within the last 24 hours
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Comment by jschipp-r7** [comment]
@r7-mgorny - for the icon files - copy the icon.png and extension.png from one of the other Cisco Umbrella plugin directories. They should all be the same

File: cisco_umbrella_reporting/plugin.spec.yaml [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+      type: "[]string"
+      required: false
+actions:
+  get_domain_visits:
+    title: Get Domain Visits
+    description: Get a list of computers that visited a domain within the last 24 hours
+    input:
+      most_recent:
+        title: Most Recent
+        description: Set to true to get the most recent DNS requests for a domain
```
Comment (jschipp-r7) [inline cisco_umbrella_reporting/plugin.spec.yaml:103]: Do we know what the difference is between setting this to true and false? The action is for last 24 hours which seems recent to me. What does most recent enable compared to not setting it?

File: cisco_umbrella_reporting/plugin.spec.yaml [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+        type: boolean
+        required: false
+        default: true
+        example: true
+      domain:
+        title: Domain
+        description: Domain to search for visits
```
Comment (jschipp-r7) [inline cisco_umbrella_reporting/plugin.spec.yaml:110]: Most Recent docs say,

> You must specify a destination. A destination can be a domain or an IP address and should be specified without any protocol or delimiters. For instance, "www.cisco.com" is valid, but "https://cisco.com" is not valid, nor is "cisco.com/index.html".

https://docs.umbrella.com/umbrella-api/docs/reporting-destinations-most-recent-requests

We will want to allow an IP address too according to the docs when Most Recent is set. We also want to allow a URL by stripping it down to a domain by removing the protocol (e.g. https://) and the suffix `/path/to/website` in this case.

File: cisco_umbrella_reporting/help.md [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+##### Output
+
+|Name|Type|Required|Description|
+|----|----|--------|-----------|
+|domain_visits|[]record|True|List of computers that visited a domain|
+
+Example output:
+
+```
+[
```
Comment (jschipp-r7) [inline cisco_umbrella_reporting/help.md:70]: This should be in a JSON object

**Review by jschipp-r7 — CHANGES_REQUESTED** [review]

File: cisco_umbrella_reporting/plugin.spec.yaml (hunk 14 above)
Comment (r7-mgorny) [reply cisco_umbrella_reporting/plugin.spec.yaml:103]: @jonschipp 
Here are used two endpoints: 
/reporting-destinations-most-recent-requests
/security-activity-report
First get 500 records in last 24 hours and later. So it should get 500 data from week in example. 
The second one get 500 records but only from 24 hours.

Comment (jschipp-r7) [reply cisco_umbrella_reporting/plugin.spec.yaml:103]: @r7-mgorny  Thanks can you document taht in the descriptions and in the help.md?

Comment (r7-mgorny) [reply cisco_umbrella_reporting/plugin.spec.yaml]: Documented.

File: cisco_umbrella_reporting/plugin.spec.yaml (hunk 15 above)
Comment (r7-mgorny) [reply cisco_umbrella_reporting/plugin.spec.yaml]: I changed input to address instead of domain. 
I tested this with URL: https://r.routemob.com:443/index.html and this works. It only get domain

File: cisco_umbrella_reporting/help.md (hunk 16 above)
Comment (r7-mgorny) [reply cisco_umbrella_reporting/help.md]: Fixed

File: cisco_umbrella_reporting/plugin.spec.yaml [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+      most_recent:
+        title: Most Recent
+        description: Set to true to get the most recent DNS requests for a domain, if set to false data will be get only for last 24h
+        type: boolean
+        required: false
+        default: true
+        example: true
+      address:
+        title: Address
+        description: Domain or IP address to search for visits
```
Comment (jschipp-r7) [inline cisco_umbrella_reporting/plugin.spec.yaml:110]:
```suggestion
        description: Domain, IP address, or URL to search for computer visits. If a URL is provided it will be converted to a domain or IP address. If this field is empty, it will return activities for all domains in the organization.
```

File: cisco_umbrella_reporting/plugin.spec.yaml [hunk 18]
```diff
@@ -0,0 +1,119 @@
...
+      required: false
+    tags:
+      title: Tags
+      description: Which tags, if any, the destination for which this request was made falls into
+      type: "[]string"
+      required: false
+actions:
+  get_domain_visits:
+    title: Get Domain Visits
+    description: Get a list of computers that visited a domain within the last 24 hours
```
Comment (jschipp-r7) [inline cisco_umbrella_reporting/plugin.spec.yaml:99]:
```suggestion
    description: Get a list of computers that visited a domain within the last 24 hours for up to 500 computers.
```

File: cisco_umbrella_reporting/plugin.spec.yaml [hunk 19]
```diff
@@ -0,0 +1,119 @@
...
+    input:
+      most_recent:
```
Comment (jschipp-r7) [inline cisco_umbrella_reporting/plugin.spec.yaml:101]: Let's remove this option, and instead set `most_recent` in the code if the `address` field is empty. This should simply for the UX.

**Review by jschipp-r7 — CHANGES_REQUESTED** [review]

File: cisco_umbrella_reporting/plugin.spec.yaml (hunk 19 above)
Comment (r7-mgorny) [reply cisco_umbrella_reporting/plugin.spec.yaml]: Done

File: cisco_umbrella_reporting/help.md [hunk 20]
```diff
@@ -0,0 +1,124 @@
...
+}
+```
+
+## Technical Details
+
+### Actions
+
+#### Get Domain Visits
+
+This action is used to get a list of computers that visited a domain for the last 24 hours.
```
Comment (jschipp-r7) [inline cisco_umbrella_reporting/help.md:43]:
```suggestion
This action is used to get a list of computers that visited a domain for the last 24 hours and up to 500 results.
It accepts the following address types: domains, IPs, and URLs as input. URLs will automatically be stripped down to the domain since the Umbrella API does not accept URLs. If no add address is provided, then the action will return activities for all domains in the organization.
```

**Review by jschipp-r7 — APPROVED** [review]
